### PR TITLE
fix failing test cases and refactor timestamp and nonce

### PR DESCRIPTION
**Pull request by shahinrahbariasl**
- Allow `NewAuthenticatedConnectionsSigner` to accept a custom random reader as param.
- Update `AuthenticatedConnectionSignatureParams` to accept custom time function in order to mock timestamp in tests.
- Refactor tests setup code in `authenticated_connections_example_test.go`.
- Fix failing tests output check.
- Remove non-determinism in key selection.
- Modify `SignAuthenticatedConnection` method logic to accept custom time object.

### CODE REVIEW

**Review by mlb7687 — DISMISSED** [review]
LGTM, but I'll let either Curtis or Josh take a look as well before merging

File: examples/signer/example-signer.go
Comment (mlb7687) [reply examples/signer/example-signer.go]: I'm guessing this is to make it clear it isn't "math/rand" which is not cryptographically secure.
This is also consistent with how Curtis had used this elsewhere.

Comment (shahinrahbariasl) [reply examples/signer/example-signer.go]: I had to namespace it in examples/signer/example-signer.go for the reason @mlb7687 mentioned because we're using "math/rand" but I can change it in the other files.

File: pkg/adscert/authenticated_connections_example_test.go
Comment (shahinrahbariasl) [reply pkg/adscert/authenticated_connections_example_test.go]: addressed.

**Review by grossjo — APPROVED** [review]

File: examples/signer/example-signer.go
Comment (grossjo) [reply examples/signer/example-signer.go]: I think the explicit crypto_rand make sense. I was just curious.

**Review by cmlight — APPROVED** [review]
LGTM

**Review by grossjo — APPROVED** [review]

**Pull request merged**
